### PR TITLE
add RR1337IfStatementInTestMethod analyzer

### PR DESCRIPTION
**Pull request by Ivasek**

### CODE REVIEW

File: src/Sorry.Analyzers/DiagnosticIds.cs [hunk 1]
```diff
@@ -5,5 +5,7 @@ internal static class DiagnosticIds
         internal const string Sorry1000 = "SORRY1000";
 
         internal const string Sorry2000 = "SORRY2000";
+
+        internal const string Sorry1337 = "SORRY1337";
```
Comment (divan-9) [inline src/Sorry.Analyzers/DiagnosticIds.cs]: How about 3000?

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+";
+
+            this.HasDiagnostic(
+                markupCode: text,
+                diagnosticId: DiagnosticIds.Sorry1337);
+        }
+
+        [Fact]
+        public void NunitTestCaseAttributeTest()
```
Comment (divan-9) [inline tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: May I kindly ask you to add some tests on negative scenarios? For example, DoesNotWarnIfInsideRegularMethod

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+}
+";
+
+            this.HasDiagnostic(
+                markupCode: text,
+                diagnosticId: DiagnosticIds.Sorry1337);
+        }
+
+        [Fact]
+        public void XunitTheoryAttributeTest()
```
Comment (divan-9) [inline tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: I used a different naming style for test methods. I don’t insist, but what if using the same naming notation is a good idea?

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
+﻿namespace Sorry.Analyzers.Tests
+{
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using RoslynTestKit;
+    using Xunit;
+
+    public class RR1337IfStatementInTestMethodTests : AnalyzerTestFixture
```
Comment (divan-9) [inline tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: Please rename testclass appropriatelly

File: src/Sorry.Analyzers/DiagnosticIds.cs (hunk 1 above)
Comment (Ivasek) [reply src/Sorry.Analyzers/DiagnosticIds.cs]: Оf course. 1337 should be the most elite rule. :-) Fixed.

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs (hunk 2 above)
Comment (Ivasek) [reply tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: Thank you for your suggestion. I added some negative test cases.

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs (hunk 3 above)
Comment (Ivasek) [reply tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: I'm so sorry. I didn't notice your naming style here. Fixed.

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs (hunk 4 above)
Comment (Ivasek) [reply tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: I renamed it. Is the current name appropriate enough?

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs (hunk 2 above)
Comment (divan-9) [reply tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: Great job. But the case when if is located inside the regular method (without any attributes) is still missing.

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs (hunk 4 above)
Comment (divan-9) [reply tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: the most appropriatest ever

File: tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs (hunk 2 above)
Comment (Ivasek) [reply tests/Sorry.Analyzers.Tests/RR1337IfStatementInTestMethodTests.cs]: Thanks. I added test case to cover it. And another one to cover non-test attribute.
